cni/pkg/taint/taint_test_helper.go

```diff
@@ -0,0 +1,211 @@
+package taint
+
+import (
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type makeConfigMapArgs struct {
+	ConfigName  string
+	Namespace   string
+	Labels      map[string]string
+	Annotations map[string]string
+	Data        map[string]string
+}
+
+const (
+	ValidationContainerName = "istio-validation"
+)
+
+func makeConfigMap(args makeConfigMapArgs) v1.ConfigMap {
+	configmap := v1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        args.ConfigName,
+			Namespace:   args.Namespace,
+			Labels:      args.Labels,
+			Annotations: args.Annotations,
+		},
+		Data: args.Data,
+	}
+	return configmap
+}
+
+type makePodArgs struct {
+	PodName             string
+	Namespace           string
+	Labels              map[string]string
+	Annotations         map[string]string
+	InitContainerName   string
+	InitContainerStatus *v1.ContainerStatus
+	Tolerations         []v1.Toleration
+	NodeName            string
+	Conditions          []v1.PodCondition
+}
+
+func makePodWithTolerance(args makePodArgs) *v1.Pod {
+	pod := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        args.PodName,
+			Namespace:   args.Namespace,
+			Labels:      args.Labels,
+			Annotations: args.Annotations,
+		},
+		Spec: v1.PodSpec{
+			NodeName: args.NodeName,
+			InitContainers: []v1.Container{
+				{
+					Name: args.InitContainerName,
+				},
+			},
+			Containers: []v1.Container{
+				{
+					Name: "payload-container",
+				},
+			},
+			Tolerations: args.Tolerations,
+		},
+		Status: v1.PodStatus{
+			Conditions: args.Conditions,
+			InitContainerStatuses: []v1.ContainerStatus{
+				*args.InitContainerStatus,
+			},
+			ContainerStatuses: []v1.ContainerStatus{
+				{
+					Name: "payload-container",
+					State: v1.ContainerState{
+						Waiting: &v1.ContainerStateWaiting{
+							Reason: "PodInitializing",
+						},
+					},
+				},
+			},
+		},
+	}
+	return pod
+}
+
+type makeNodeArgs struct {
+	NodeName      string
+	Taints        []v1.Taint
+	NodeCondition []v1.NodeCondition
+}
+
+func makeNodeWithTaint(args makeNodeArgs) *v1.Node {
+	node := &v1.Node{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Node",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: args.NodeName,
+		},
+		Spec: v1.NodeSpec{
+			Taints: args.Taints,
+		},
+		Status: v1.NodeStatus{
+			Conditions: args.NodeCondition,
+		},
+	}
+	return node
+}
+
+var (
+	//Data for configMaps
+	istiocniConfig = makeConfigMap(makeConfigMapArgs{
+		ConfigName: "node.readiness",
+		Namespace:  "kube-system",
+		Data: map[string]string{
+			"istio-cni.properties": `name: istio-cni
+selector: app=istio 
+namespace: kube-system`,
+		},
+	})
+	combinedConfig = makeConfigMap(makeConfigMapArgs{
+		ConfigName: "node.readiness",
+		Namespace:  "kube-system",
+		Data: map[string]string{
+			"istio-cni.properties": `name: istio-cni
+selector: app=istio
+namespace: kube-system`,
+			"others.properties": `name: others
+selector: app=others	
```

Comment:
do we support advanced selectors like `app=other,!foo`? should we have a test somewhere?